### PR TITLE
Deprecate /api/devfile REST API

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Deprecate /api/devfile REST API

### What issues does this PR fix or reference?
need for https://github.com/eclipse/che/issues/13867

#### Release Notes
n/a

#### Docs PR
n/a

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ci-test

**Review by metlos — COMMENTED** [review]
I'm generally OK with returning 400 with a meaningful error message but I think we should consider the following:

* we should remove the swagger docs from the methods
* what about returning 301 (or maybe even 308) instead?

**Comment by skabashnyuk** [comment]
>we should remove the swagger docs from the methods

ok

>what about returning 301 (or maybe even 308) instead?

My intention was a bit different. It's not just to keep it working it's to identify a place what it used and fix it. I next release we want to remove this api at all because we have analog ```/api/workspace/devfile```. 

AFAIK there are no places where we use this api.

**Comment by skabashnyuk** [comment]
@rhopp @nickboldt @sleshchenko @mshaposhnik thoughts?

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/DevfileService.java [hunk 1]
```diff
@@ -124,16 +119,8 @@ public Response createFromYaml(String data)
...
-      DevfileImpl devfile = devfileManager.parseYaml(data);
-      workspace = devfileManager.createWorkspace(devfile, urlFileContentProvider);
-    } catch (DevfileException e) {
-      throw new BadRequestException(e.getMessage());
-    }
-    return Response.status(201)
-        .entity(asDto(workspace).withLinks(linksGenerator.genLinks(workspace, getServiceContext())))
-        .build();
+    throw new BadRequestException(
+        "This is an outdated method please use /api/workspace/devfile to create a new workspace");
```
Comment (nickboldt) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/DevfileService.java]: Run-on sentence; does not explain WHY user should use other api.

Preferred: "This method is deprecated and may be removed at any time. Please use ..."

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/DevfileService.java [hunk 2]
```diff
@@ -166,9 +153,7 @@ public Response createFromWorkspace(
           String key)
       throws NotFoundException, ServerException, BadRequestException, ConflictException,
           JsonProcessingException {
-    validateKey(key);
-    return Response.ok()
-        .entity(objectMapper.writeValueAsString(devfileManager.exportWorkspace(key)))
-        .build();
+    throw new BadRequestException(
+        "This is an outdated method please use /api/workspace/devfile to create a new workspace");
```
Comment (nickboldt) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/DevfileService.java]: Run-on sentence; does not explain WHY user should use other api.

Preferred: "This method is deprecated and may be removed at any time. Please use ..."

**Review by nickboldt — CHANGES_REQUESTED** [review]
Please tweak the error messages to explain the need for the call to action, and to fix the English syntax.

**Comment by skabashnyuk** [comment]
@nickboldt your comments have been addressed

**Comment by skabashnyuk** [comment]
ci-test

**Review by rhopp — APPROVED** [review]

**Comment by skabashnyuk** [comment]
@nickboldt are you ok with the current state of pr?
@eclipse/eclipse-che-qa can I ignore results of crw-ci-e2e-happy-path-test ?

**Comment by SkorikSergey** [comment]
> @eclipse/eclipse-che-qa can I ignore results of crw-ci-e2e-happy-path-test ?

Yes, it is test problem(20 sec was not enough for starting Java language server).

**Comment by skabashnyuk** [comment]
@nickboldt ping

**Review by nickboldt — APPROVED** [review]
+1

**Comment by nickboldt** [comment]
@skabashnyuk sorry for slow response. You'd do better to ping on MM, slack, IRC, hangouts or anything that doesn't produce email but a notification of unread messages. :) 

Email is the thing I do LAST after all the other hailing frequencies are done.

**Pull request merged**